packages/js/src/preferences/helpers.ts

```diff
@@ -0,0 +1,64 @@
+import { ApiService } from '@novu/client';
+
+import type { NovuEventEmitter } from '../event-emitter';
+import type { ChannelPreferenceOverride, TODO } from '../types';
+import { PreferenceLevel } from '../types';
+import { Preference } from './preference';
+import type { UpdatePreferencesArgs } from './types';
+
+export const mapPreference = (apiPreference: {
+  template?: TODO;
+  preference: {
+    enabled: boolean;
+    channels: {
+      email?: boolean;
+      sms?: boolean;
+      in_app?: boolean;
+      chat?: boolean;
+      push?: boolean;
+    };
+    overrides?: ChannelPreferenceOverride[];
+  };
+}): Preference => {
+  const { template: workflow, preference } = apiPreference;
+  const hasWorkflow = workflow !== undefined;
+  const level = hasWorkflow ? PreferenceLevel.TEMPLATE : PreferenceLevel.GLOBAL;
+
+  return new Preference({
+    level,
+    enabled: preference.enabled,
+    channels: preference.channels,
+    workflow,
+    overrides: preference.overrides,
+  });
+};
```

Comment:
Map server preference to the one that we use in the module with a better interface.